### PR TITLE
Adding tweets to storybook with stubbed data

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/components/Tweets/Tweet.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import styles from './Tweet.scss';
+
+const Tweets = (props) => {
+  return (
+      <div className={cx(styles.block)}>
+        <div className={cx(styles.image)}>
+          <img width="48" height="48" src={props.user.profile_background_image_url}/>
+        </div>
+        <div className={cx(styles.tweet)}>
+          <li><span className={cx(styles.fullName)}>{props.user.name}</span> - <span className={cx(styles.screenName)}>@{props.user.screen_name}</span></li>
```
Comment (ghost) [inline src/components/Tweets/Tweet.js]: I'll sort the missing <ul> tags out.

File: src/components/Tweets/Tweet.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import React, { PropTypes } from 'react';
+import cx from 'classnames';
+
+import styles from './Tweet.scss';
+
+const Tweets = (props) => {
+  return (
+      <div className={cx(styles.block)}>
+        <div className={cx(styles.image)}>
+          <img width="48" height="48" src={props.user.profile_background_image_url}/>
```
Comment (michael-martin) [inline src/components/Tweets/Tweet.js]: Can you add a config.js file in /src/ which exports just a simple object? Then you can move avatarSize as a parameter in there, so this value is configurable.

File: src/components/Tweets/Tweet.js (hunk 1 above)
Comment (michael-martin) [inline src/components/Tweets/Tweet.js]: Thanks :)

You also don't need the cx() calls here. cx() lets you add multiple classes at once but if you're just using one class, it's fine to write `className={styles.fullName}` 

Can you split this line onto multiple lines as well? When eslint kicks in again, the 100 character line limit will kick in too.

File: src/components/Tweets/Tweet.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        </div>
+        <div className={cx(styles.tweet)}>
+          <li><span className={cx(styles.fullName)}>{props.user.name}</span> - <span className={cx(styles.screenName)}>@{props.user.screen_name}</span></li>
+          <li>{props.text}</li>
+        </div>
+      </div>
+  );
+};
```
Comment (michael-martin) [inline src/components/Tweets/Tweet.js]: Can you add PropTypes validation here as well? Check out my CardView PR for an example

**Comment by michael-martin** [comment]
I've left a few comments now, but 2 more structural things:
- The overall folder name is plural, but the component itself is singular. Can the folder be renamed to TweetView? (The "View" suffix is just a pattern to distinguish presentational components)
- The Tweet takes a full object of Twitter data right now, which means the Tweet component needs to know the structure of the Twitter API, and we're passing it a tonne of properties that it doesn't need.

With these Views, it's best to give them only what they need, and be very explicit about it. That may mean that your container component when added later picks out the parts of the API response that matter, or even that an API layer may do that.

For this View PR though, we don't need to solve that. Instead, I'd remove the API response currently in the story, and update the component and story to only accept/pass the individual pieces of data needed, i.e. screenName etc. should be props

File: src/components/TweetView/Tweet.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import React, { PropTypes } from 'react';
+import cx from 'classnames';
+import config from '../../config.js';
+import styles from './Tweet.scss';
+
+const Tweets = (props) => {
+  return (
+    <div className={styles.block}>
+      <div className={styles.image}>
+        <img width={config.avatarSize} height={config.avatarSize} src={props.image}/>
```
Comment (vitkon) [inline src/components/TweetView/Tweet.js:10]: can we add `alt` attribute? it's required for accessibility

possibly a better name for `image` will be `imageUrl`

File: src/components/TweetView/Tweet.scss [hunk 5]
```diff
@@ -0,0 +1,36 @@
+.block {
+  border-bottom: 1px solid #e1e8ed;
+  border-left: 1px solid #e1e8ed;
+  border-right: 1px solid #e1e8ed;
+  padding: 10px 15px;
+  list-style: none;
+  background-color: white;
+  &:hover {
+    background-color: #f5f8fa;
```
Comment (vitkon) [inline src/components/TweetView/Tweet.scss:9]: It'd be good to move all colour swatches into a separate _variables_ sass file to introduce some consistency with the palette.

File: src/components/TweetView/Tweet.scss [hunk 6]
```diff
@@ -0,0 +1,36 @@
+.block {
+  border-bottom: 1px solid #e1e8ed;
+  border-left: 1px solid #e1e8ed;
+  border-right: 1px solid #e1e8ed;
+  padding: 10px 15px;
```
Comment (vitkon) [inline src/components/TweetView/Tweet.scss:5]: can we use relative units for padding? (especially for vertical one).
That will help us to maintain a vertical rhythm even on screens with zoom enabled.

File: src/components/TweetView/Tweet.scss [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+}
+
+.tweet {
+  display: inline-block;
+  vertical-align: top;
+  width: 90%;
+  li {
+    list-style-type: none;
+    .screenName {
+      font-size: 13px;
```
Comment (vitkon) [inline src/components/TweetView/Tweet.scss:20]: same for the font-size values — they all need to live in a _variables_ sass for consistency

File: src/components/TweetView/Tweet.js [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+        </ul>
+      </div>
+    </div>
+  );
+};
+
+Tweets.propTypes = {
+  userName: React.PropTypes.func.isRequired,
+  doubleAsync: React.PropTypes.func.isRequired,
+  increment: React.PropTypes.func.isRequired
```
Comment (michael-martin) [inline src/components/TweetView/Tweet.js:28]: PropTypes is already imported as its own object on line 1, so can delete the `React.` from each of these lines

File: src/components/TweetView/Tweet.scss (hunk 5 above)
Comment (michael-martin) [inline src/components/TweetView/Tweet.scss:9]: @victoriasloan is working on this, will come through in a separate PR.

@girwin575 Can you make the quick imageUrl/alt/React. changes and then we'll merge this? Means @victoriasloan can base the variables update off master then, which will have the 2 current View PRs merged, so all CSS is covered.

**Pull request closed without merging**